diffprivlib/models/tree.py

```diff
@@ -0,0 +1,602 @@
+        The number of features when fit is performed.
+
+    n_classes: int
+        The number of classes.
+
+    classes_: array of shape (n_classes, )
+        The classes labels.
+
+    cat_features: array of categorical feature indexes
+        Categorical feature indexes.
+
+    max_depth: int
+        Final max depth used for constructing decision trees.
+
+    estimators: list of DecisionTreeClassifier
+        The collection of fitted sub-estimators.
+
+    feature_domains: dictionary of domain values mapped to feature
+        indexes in the training data
+
+    Examples
+    --------
+    >>> from sklearn.datasets import make_classification
+    >>> from diffprivlib.models.tree import RandomForestClassifier
+    >>> X, y = make_classification(n_samples=1000, n_features=4,
+    ...                            n_informative=2, n_redundant=0,
+    ...                            random_state=0, shuffle=False)
+    >>> clf = RandomForestClassifier(n_estimators=100, random_state=0)
+    >>> clf.fit(X, y)
+    >>> print(clf.predict([[0, 0, 0, 0]]))
+    [1]
+
+    References
+    ----------
+    [1] Sam Fletcher, Md Zahidul Islam. "Differentially Private Random Decision Forests using Smooth Sensitivity"
+    https://arxiv.org/abs/1606.03572
+
+    """
+
+    def __init__(self, n_estimators=10, epsilon=1.0, cat_feature_threshold=10,
+                 n_jobs=1, verbose=0, accountant=None, max_depth=15, random_state=None, feature_domains=None, **unused_args):
+        super().__init__(
+            base_estimator=DecisionTreeClassifier(),
+            n_estimators=n_estimators,
+            estimator_params=("cat_feature_threshold", "max_depth", "epsilon", "random_state"),
+            n_jobs=n_jobs,
+            random_state=random_state,
+            verbose=verbose)
+        self._n_estimators = n_estimators
+        self._n_jobs = n_jobs
+        self._epsilon = epsilon
+        self._cat_feature_threshold = cat_feature_threshold
+        self._random_state = random_state
+        self._estimators = []
+        self._verbose = verbose
+        self._max_depth = max_depth
+        self._accountant = BudgetAccountant.load_default(accountant)
+        self._n_features = None
+        self._classes = None
+        self._cat_features = None
+        self._feature_domains = feature_domains
+
+        if random_state is not None:
+            np.random.seed(random_state)
+
+        warn_unused_args(unused_args)
+
+    @property
+    def n_features(self):
+        return self._n_features
+
+    @property
+    def n_classes(self):
+        return self._n_classes
+
+    @property
+    def classes_(self):
+        return self._classes
+
+    @property
+    def cat_features(self):
+        return self._cat_features
+
+    @property
+    def max_depth(self):
+        return self._max_depth
+
+    @property
+    def estimators(self):
+        return self._estimators
+
+    @property
+    def feature_domains(self):
+        return self._feature_domains
```

Comment:
These need not be defined, as they can be accessed directly from the class variable.